File: this-is-coding-test/4.구현/1.1.상하좌우/1.py
# n = int(input())
# plans = input().split()
n = 5
plans = ['R', 'R', 'R', 'U', 'D', 'D']
x, y = 1, 1

# direction = dict()
# [direction['U'], direction['D'], direction['L'], direction['R']] = [[-1, 0], [1, 0], [0, -1], [0, 1]]
direction = {
    'U': [-1, 0],
    'D': [1, 0],
    'L': [0, -1],
    'R': [0, 1],
}

for plan in plans:
    [dx, dy] = direction[plan]
    [nx, ny] = [x+dx, y+dy]

    if 1 <= nx <= n and 1 <= ny <= n:
        x, y = nx, ny

print(x, y)

File: this-is-coding-test/4.구현/1.1.상하좌우/1.js
const limit = 5;
const init = {
  row: 1,
  col: 1,
};

const plans = ["R", "R", "R", "U", "D", "D"];

const direction = {
  U: { row: -1, col: 0 },
  D: { row: 1, col: 0 },
  R: { row: 0, col: 1 },
  L: { row: 0, col: -1 },
};

const result = plans.reduce((acc, cur) => {
  console.log(acc, cur);
  const row = acc.row + direction[cur].row;
  const col = acc.col + direction[cur].col;

  if (0 >= row || 0 >= col || row > 5 || col > 5) {
    return acc;
  } else {
    return {
      row,
      col,
    };
  }
}, init);

console.log(result);
